sdk/js/src/api/index.js

```diff
@@ -62,10 +59,15 @@ class Api {
             route = route.replace(`{${parameter}}`, routeParams[parameter])
           }
 
-          const searchQuery = Marshaler.query(queryParams)
-          route = searchQuery ? `${route}?${searchQuery}` : route
+          if (endpoint.method === 'delete') {
+            return connector.delete(route)
+          }
+
+          if (endpoint.method === 'get') {
+            return connector.get(route, { queryParams, fieldMask })
+          }
 
-          return connector[endpoint.method](route, body)
+          return connector[endpoint.method](route, payload, { queryParams, fieldMask })
```

Comment:
arent query parameters only for get requests?

Reply:
No, they can be used always. Though we probably won't face that in the API.